src/test/end.t.sol

```diff
@@ -483,6 +504,101 @@ contract EndTest is DSTest {
         assertEq(balanceOf("gold", address(gold.gemA)), 0);
     }
 
+    // -- Scenario where there is one collateralised CDP
+    // -- undergoing auction at the time of cage
+    function test_cage_snip() public {
+        Ilk memory gold = init_collateral("gold");
+
+        Usr ali = new Usr(vat, end);
+
+        vat.fold("gold", address(vow), int256(ray(0.25 ether)));
+
+        // Make a CDP:
+        address urn1 = address(ali);
+        gold.gemA.join(urn1, 10 ether);
+        ali.frob("gold", urn1, urn1, urn1, 10 ether, 15 ether);
+        (uint ink1, uint art1) = vat.urns("gold", urn1); // CDP before liquidation
+        (, uint rate,,,) = vat.ilks("gold");
+
+        assertEq(vat.gem("gold", urn1), 0);
+        assertEq(rate, ray(1.25 ether));
+        assertEq(ink1, 10 ether);
+        assertEq(art1, 15 ether);
+
+        vat.file("gold", "spot", ray(1 ether)); // Now unsafe
+
+        uint256 id = dog.bark("gold", urn1);   
+
+        uint256 tab1;
+        uint256 lot1;    
+        {
+            address usr1;
+            uint96  tic1;
+            uint256 top1; 
+            uint256 pos1;
+            (pos1, tab1, lot1, usr1, tic1, top1) = gold.clip.sales(id);  
+            assertEq(pos1, 0);
+            assertEq(tab1, art1 * rate * 1.1 ether / WAD); // tab uses chop
+            assertEq(lot1, ink1);
+            assertEq(usr1, address(ali));
+            assertEq(uint256(tic1), now);
+            assertEq(uint256(top1), ray(6 ether));
+        }
+                     
+        assertEq(dog.Dirt(), tab1);
+
+        {
+            (uint ink2, uint art2) = vat.urns("gold", urn1); // CDP after liquidation
+            assertEq(ink2, 0);
+            assertEq(art2, 0);
+        }
+
+        // Collateral price is $5
+        gold.pip.poke(bytes32(5 * WAD));
+        spot.poke("gold");
+        end.cage();
+        end.cage("gold");
+        assertEq(end.tag("gold"), ray(0.2 ether)); // par / price = collateral per DAI
+
+        assertEq(vat.gem("gold", address(gold.clip)), lot1); // From grab in dog.bark()
+        assertEq(vat.sin(address(vow)),        art1 * rate); // From grab in dog.bark()
+        assertEq(vat.vice(),                   art1 * rate); // From grab in dog.bark()
+        assertEq(vat.debt(),                   art1 * rate); // From frob
+        assertEq(vat.dai(address(vow)),                  0); // vat.suck() hasn't been called 
+
+        end.snip("gold", id);
+
+        uint256 tab2;
+        uint256 lot2;    
```

Comment:
Any reason not to move these inside the scoping block below?

Reply:
Good point I'll move them in, I copied this block from the first check where I needed `tab1` and `lot1` out of scope.